### ISSUE TITLE
Unable to get Current ID (mode), as APPS did not exist anymore

### ISSUE THREAD

**Issue by MattiasBiri** [v1ack/smartLightConnector#3]
Hi!

Would really like to use your integration,
but when I tried to get my _Current ID (mode)_ I have failed.

The mentioned apps did not exist anymore.

I managed to control my lamp with app, "Smartcontrol".
But that app shows no Current ID. It shows a QR code to share access only.

Is there any up-to-date method to get the Current ID?
Maybe you have the GM Light app apk? I was not able to google it.
I only found GM Light  v0.0.3 but that does not offer pairing.

Stuck here :)

**Comment by amrugal** [comment]
Old issue but maybe someone will find it helpful. I have app downloaded from instruction delivered with lamp (paper form). It is GM Light v0.0.3r_03041358 and I have it on my phone(apk file). I think this app version can pair it but it is tricky...
Turn off lamp from electricity and be prepared to tap "ON" in the app... Connect lamp to electricity and after it will shine tap mentioned "ON" button immediately. It should be paired and it will be confirmed by pulsing for a sec.

**Comment by amrugal** [comment]
> Hi!
> 
> Would really like to use your integration, but when I tried to get my _Current ID (mode)_ I have failed.
> 
> The mentioned apps did not exist anymore.
> 
> I managed to control my lamp with app, "Smartcontrol". But that app shows no Current ID. It shows a QR code to share access only.
> 
> Is there any up-to-date method to get the Current ID? Maybe you have the GM Light app apk? I googled for archived APKs, but have not found any. I only found GM Light v0.0.3 but that is so early, that it does not offer pairing.
> 
> Stuck here :)

Old issue but maybe someone will find it helpful. I have app downloaded from instruction delivered with lamp (paper form). It is GM Light v0.0.3r_03041358 and I have it on my phone(apk file). I think this app version can pair it but it is tricky...
Turn off lamp from electricity and be prepared to tap "ON" in the app... Connect lamp to electricity and after it will shine tap mentioned "ON" button immediately. It should be paired and it will be confirmed by pulsing for a sec.

**Comment by caideone** [comment]
I have been researching lemesh recently，I have many gateways provided by manufacturers that can achieve remote control, but I have been unable to access Homeassistant

**Comment by caideone** [comment]
> 你好！
> 
> 真的很想使用您的集成， 但是当我尝试获取我的_当前ID（模式）_时，我失败了。
> 
> 前面提到的应用程序已不再存在。
> 
> 我设法用应用程序“Smartcontrol”控制我的灯。 但该应用程序没有显示当前ID。它显示一个二维码，可以共享访问。
> 
> 有没有最新的方法可以获取当前ID？ 也许你有GM Light应用程序apk？我在google上搜索文档的APK，但没有找到。 我只找到了GM Light v0.0.3，但那太早了，它没有提供码头。
> 
> 卡在这里:)

你有没有搞定

**Comment by amrugal** [comment]
> > 你好！
> > 真的很想使用您的集成， 但是当我尝试获取我的_当前ID（模式）_时，我失败了。
> > 前面提到的应用程序已不再存在。
> > 我设法用应用程序“Smartcontrol”控制我的灯。 但该应用程序没有显示当前ID。它显示一个二维码，可以共享访问。
> > 有没有最新的方法可以获取当前ID？ 也许你有GM Light应用程序apk？我在google上搜索文档的APK，但没有找到。 我只找到了GM Light v0.0.3，但那太早了，它没有提供码头。
> > 卡在这里:)
> 
> 你有没有搞定

https://github.com/amrugal/gm_light_v3/blob/main/gm_0.0.3r_03041358.apk

**Comment by caideone** [comment]
> > > 你好！
> > > 真的很想使用您的集成，但是当我尝试获取我的_当前ID（模式）_时，我失败了。
> > > 前面提到的应用程序已不再存在。
> > > 我设法使用应用程序“Smartcontrol” ”控制我的灯。但是该应用程序没有显示当前ID。它显示一个二维码，可以共享访问。有
> > > 没有最新的方法可以获取当前ID？你也许有GM灯应用程序apk？我在google上搜索文档的APK，但没有找到。我只找到了GM Light v0.0.3，但那太早了，它没有提供码头。
> > > 卡在这里:)
> > 
> > 
> > 你有没有搞定
> 
> https://github.com/amrugal/gm_light_v3/blob/main/gm_0.0.3r_03041358.apk

我意思 你有没有试试他的homeassistant hacs 插件 ，在home assistant 控制light

**Comment by amrugal** [comment]
> > > > 你好！
> > > > 真的很想使用您的集成，但是当我尝试获取我的_当前ID（模式）_时，我失败了。
> > > > 前面提到的应用程序已不再存在。
> > > > 我设法使用应用程序“Smartcontrol” ”控制我的灯。但是该应用程序没有显示当前ID。它显示一个二维码，可以共享访问。有
> > > > 没有最新的方法可以获取当前ID？你也许有GM灯应用程序apk？我在google上搜索文档的APK，但没有找到。我只找到了GM Light v0.0.3，但那太早了，它没有提供码头。
> > > > 卡在这里:)
> > > 
> > > 
> > > 你有没有搞定
> > 
> > 
> > https://github.com/amrugal/gm_light_v3/blob/main/gm_0.0.3r_03041358.apk
> 
> 我意思 你有没有试试他的homeassistant hacs 插件 ，在home assistant 控制light

No. This integration does not work on my PC. I have Home assistant virtual machine in proxmox. I think integration works on raspberry pi because author was developing it on raspberry pi, tested it and it work for him.

**Comment by caideone** [comment]
好吧 我晚会用树莓派测试下，之前我用esxi虚拟机测试，灯没有反映

**Comment by v1ack** [comment]
Hello
Yes, tested on raspberry pi and linux
Only works if Bluetooth adapter available for home assistant

Virtual machine doesn't work for me, looks like home assistant must be installed on host os, or docker with privileged mode (linux host only)

Yes, the latest version of app is 0.0.3
